src/transformState.js

```diff
@@ -5,7 +5,46 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear': {
+        toClear();
+        break;
+      }
+
+      case 'removeProperties': {
+        toRemove(action.keysToRemove);
+        break;
+      }
+
+      case 'addProperties': {
+        toAdd(action.extraData);
+        break;
+      }
+
+      default: {
+        window.alert('no similar data find');
+      }
+    }
+  }
+
+  function toClear() {
+    for (const key in state) {
+      delete state[key];
+    }
+  }
+
+  function toRemove(x) {
```

Comment:
you should never name your function with just one letter. Variable should describe what is stored inside